src/ssl_asn1.c

```diff
@@ -282,10 +284,13 @@ static int wolfssl_i2d_asn1_items(const void* obj, byte* buf,
             len = 0;
             break;
         }
+
         if (buf != NULL && tmp != NULL && !mem->ex && mem->tag >= 0) {
-            /* Encode the implicit tag */
             byte imp[ASN_TAG_SZ + MAX_LENGTH_SZ];
-            SetImplicit(tmp[0], mem->tag, 0, imp, 0);
+            /* Encode the implicit tag; There's other stuff in the upper bits
+             * of the integer tag, so strip out everything else for value. */
+            SetImplicit(tmp[0], (byte)(mem->tag & ASN_IMPLICIT_TAG_MASK),
```

Comment:
Indeed there's no need to _explicitly_ mask, and I suppose with the new comments it's ok. But I generally prefer code that's more intuitive to immediately understand. That's also why I spelled out the full size of `0x000000FF` and not just `0xFF`. Happy to change it if you prefer otherwise.

Reply:
I would prefer the masking removed and it suggests that not all of the bottom 8 bits will be used. We want it all.

Reply:
ok, @SparkiDev - masking macro removed.